sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update.hpp

```diff
@@ -0,0 +1,364 @@
+//
+//===----------------------------------------------------------------------===//
+
+#include "../../esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+#ifdef USE_64_BIT_OFFSET
+typedef uint64_t Toffset;
+#else
+typedef uint32_t Toffset;
+#endif
+
+constexpr int Signed = 1;
+constexpr int Unsigned = 2;
+
+struct Config {
+  int64_t threads_per_group;
+  int64_t n_groups;
+  int64_t start_ind;
+  int64_t masked_lane;
+  int64_t repeat;
+  int64_t stride;
+};
+
+// ----------------- Helper functions
+
+std::ostream &operator<<(std::ostream &out, const Config &cfg) {
+  out << "{ thr_per_group=" << cfg.threads_per_group
+      << " n_groups=" << cfg.n_groups << " start_ind=" << cfg.start_ind
+      << " masked_lane=" << cfg.masked_lane << " repeat=" << cfg.repeat
+      << " stride=" << cfg.stride << " }";
+  return out;
+}
+
+const char *to_string(atomic_op op) {
+  switch (op) {
+  case atomic_op::add:
+    return "add";
+  case atomic_op::sub:
+    return "sub";
+  case atomic_op::inc:
+    return "inc";
+  case atomic_op::dec:
+    return "dec";
+  case atomic_op::umin:
+    return "umin";
+  case atomic_op::umax:
+    return "umax";
+  case atomic_op::xchg:
+    return "xchg";
+  case atomic_op::cmpxchg:
+    return "cmpxchg";
+  case atomic_op::bit_and:
+    return "bit_and";
+  case atomic_op::bit_or:
+    return "bit_or";
+  case atomic_op::bit_xor:
+    return "bit_xor";
+  case atomic_op::smin:
+    return "smin";
+  case atomic_op::smax:
+    return "smax";
+  case atomic_op::fmax:
+    return "fmax";
+  case atomic_op::fmin:
+    return "fmin";
+  case atomic_op::fadd:
+    return "fadd";
+  case atomic_op::fsub:
+    return "fsub";
+  case atomic_op::fcmpxchg:
+    return "fcmpxchg";
+  case atomic_op::load:
+    return "load";
+  case atomic_op::store:
+    return "store";
+  case atomic_op::predec:
+    return "predec";
+  }
+  return "<unknown>";
+}
+
+template <int N> inline bool any(simd_mask<N> m, simd_mask<N> ignore_mask) {
+  simd_mask<N> m1 = 0;
+  m.merge(m1, ignore_mask);
+  return m.any();
+}
+
+// ----------------- The main test function
+template <class T, int N, template <class, int> class ImplF, bool UseProperties>
+bool test(queue q, const Config &cfg) {
+  constexpr auto op = ImplF<T, N>::atomic_op;
+  using CurAtomicOpT = decltype(op);
+  constexpr int n_args = ImplF<T, N>::n_args;
+
+  std::cout << "USM Testing " << "op=" << to_string(op) << "n_args=" << n_args
+            << " T=" << esimd_test::type_name<T>() << " N=" << N << "\n\t"
+            << cfg << "...";
+
+  size_t size = cfg.start_ind + (N - 1) * cfg.stride + 1;
+  T *arr = malloc_shared<T>(size, q);
+  int n_threads = cfg.threads_per_group * cfg.n_groups;
+
+  for (int i = 0; i < size; ++i) {
+    arr[i] = ImplF<T, N>::init(i, cfg);
+  }
+
+  range<1> glob_rng(n_threads);
+  range<1> loc_rng(cfg.threads_per_group);
```

Comment:
this is kind of a nit but the code style here doesn't match what we've been doing for the other sets, we've been doing variables names like `LocRng` instead of `loc_rng`, but I'm not sure if it's really worth rewriting the test at this point

Reply:
I'd suggest to rename everything when the porting is completed. It would be easier to reference the original (quite confusing) test.